uat/src/main/java/com.aws.greengrass/FileSteps.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass;
+
+import com.aws.greengrass.testing.model.ScenarioContext;
+import com.aws.greengrass.testing.model.TestContext;
+import com.aws.greengrass.testing.platform.Platform;
+import com.google.inject.Inject;
+import io.cucumber.guice.ScenarioScoped;
+import io.cucumber.java.en.And;
+import org.apache.commons.text.RandomStringGenerator;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+@ScenarioScoped
+public class FileSteps {
+    private final Platform platform;
+    private final TestContext testContext;
+    private final ScenarioContext senarioContext;
+    private static Logger LOGGER = LogManager.getLogger(FileSteps.class);
+    public Map<String, List<String>> msgMap=new HashMap<String, List<String>>();
+    private static final RandomStringGenerator RANDOM_STRING_GENERATOR =
+            new RandomStringGenerator.Builder().withinRange('a', 'z').build();
+    @Inject
+    public FileSteps(Platform platform, TestContext testContext, ScenarioContext senarioContext) {
+        this.platform = platform;
+        this.testContext = testContext;
+        this.senarioContext= senarioContext;
+    }
+    /**
+     * Arranges some log files with content on the /logs folder for a component
+     * to simulate a devices where logs have already bee written.
+     * @param numFiles       number of log files to write.
+     * @param componentName  name of the component.
+     * @throws IOException   thrown when file fails to be written.
+     */
+    @And("{int} temporary rotated log files for component {word} have been created")
+    public void arrangeComponentLogFiles(int numFiles, String componentName) throws IOException {
+        Path logsDirectory = testContext.installRoot().resolve("logs");
+        LOGGER.info("Writing {} log files into {}", numFiles, logsDirectory.toString());
+        if (!platform.files().exists(logsDirectory)) {
+            throw new IllegalStateException("No logs directory");
+        }
+        senarioContext.put(componentName + "LogDirectory",logsDirectory.toString());
+        if (componentName.equals("aws.greengrass.Nucleus")) {
+            msgMap.clear();
+            for (int i = 0; i < numFiles; i++) {
+                String fileName = String.format("greengrass_%s.log", i);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: i (declared at line 59). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 60. Consider formatting this data with &#39;%d&#39; instead.